src/configure/configure.ts

```diff
@@ -0,0 +1,246 @@
+    }
+}
+
+async function getAzureDevOpsDetails(): Promise<void> {
+    // TODO: handle space in project name and repo name.
+    if (!extensionVariables.azureDevOpsService.getOrganizationName()) {
+        let organizationList: string[] = await extensionVariables.azureDevOpsService.listOrganizations();
+        let selectedOrganization = await extensionVariables.uiExtensionVariables.ui.showQuickPick(organizationList.map((org) => { return { label: org }; }), { placeHolder: "Select Azure DevOps Organization" });
+        extensionVariables.azureDevOpsService.setOrganizationName(selectedOrganization.label);
+        extensionVariables.inputs.organizationName = selectedOrganization.label;
+    }
+    else {
+        extensionVariables.inputs.organizationName = extensionVariables.azureDevOpsService.getOrganizationName();
+    }
+
+    if (!extensionVariables.azureDevOpsService.getProjectName()) {
+        let projectList = await extensionVariables.azureDevOpsService.listProjects();
+        let selectedProject = await extensionVariables.uiExtensionVariables.ui.showQuickPick(projectList.map((project) => { return { label: project }; }), { placeHolder: "Select Azure DevOps project" });
+        extensionVariables.azureDevOpsService.setProjectName(selectedProject.label);
+        extensionVariables.inputs.projectName = selectedProject.label;
+    }
+    else {
+        extensionVariables.inputs.projectName = extensionVariables.azureDevOpsService.getProjectName();
+    }
+}
+
+async function getSelectedPipeline(): Promise<void> {
+    let appropriatePipelines: string[] = await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: "Analyzing your repo" }, () => {
+        return analyzeRepoAndListAppropriatePipeline(extensionVariables.inputs.sourceRepositoryDetails.localPath);
+    });
+
+    // TO:DO- Get applicable pipelines for the repo type and azure target type if target already selected
+    let selectedOption = await extensionVariables.uiExtensionVariables.ui.showQuickPick(appropriatePipelines.map((pipeline) => { return { label: pipeline }; }), {
+        placeHolder: "Select Azure pipelines template..."
+    });
+    extensionVariables.inputs.selectedPipeline = selectedOption.label;
+}
+
+async function getAzureResourceDetails() {
+    let subscriptions: [{session: any, subscription: SubscriptionModels.Subscription}] = extensionVariables.azureAccountApi.exports.subscriptions;
+    let subscriptionList = subscriptions.map((subscriptionObject) => {
+        return <QuickPickItem>{
+            label: <string>subscriptionObject.subscription.displayName
+        };
+    });
+    let selectedSubscription: QuickPickItem = await extensionVariables.uiExtensionVariables.ui.showQuickPick(subscriptionList, { placeHolder: "Select Azure Subscription" });
+    extensionVariables.inputs.subscriptionId = subscriptions.find((subscriptionObject) => {
```

Comment:
Can we add value with label in QuickPickItem ? If so, we  can add subscriptionId as value in the list and avoid searching using name.

Reply:
We can add extra information as either
- Detail
- Description
But these are visible to the user, and thus PM asked it to be removed.